### PR TITLE
Document config file options

### PR DESCRIPTION
**Pull request by bobturneruk**

### CODE REVIEW

**Review by ns-rse — COMMENTED** [review]
One issue with regards to default behaviour which hasn't yet been implemented but I think it should be, otherwise looks good. Thanks @bobturneruk

File: docs/configuration.md [hunk 1]
```diff
@@ -1,9 +1,78 @@
 ## Parameter Configuration
 
-*todo: document YAML config.*
+The parameters that the software uses for analysis of the data can be configured in a [YAML](https://yaml.org/) file. An example configuration file is included in the directory `config/example.yml`. You do not need to edit the code to change the parameters.
 
-The parameters that the software uses for analysis of the data can be configured in `Config.ini` by simply opening the file in a text editor and changing the variables. You do not need to edit the code to change the parameters.
+If no configuration YAML file is found while running TopoStats, it will generate a default file.
```
[the remark this answers is not in the document]
Comment (bobturneruk) [reply docs/configuration.md:5]: Cheers @ns-rse - helps with my understanding of what the software does, too.

**Comment by bobturneruk** [comment]
This is ready for some feedback. There are a few TBDs which by pooling the team's knowledge we should be able to fill in. And I've made some assumptions which are probably wrong. If anyone would like to take this branch and edit it directly, please feel free to, but leave a comment here so we know who has it.

File: docs/configuration.md [hunk 2]
```diff
@@ -1,9 +1,92 @@
...
+  save: True
+  colorbar: True
+  cmap: nanoscope # Options : nanoscope, afmhot
+```
+
+### Core parameters
+
+| Parameter | Description | Example | Allowable |
+| --- | --- | --- | --- |
+| `base_dir` | The directory (folder) in which the software should search for images to analyse. | `./tests/` | Path. |
```
Comment (MaxGamill-Sheffield) [inline docs/configuration.md:51]: maybe add "recursively search"

File: docs/configuration.md [hunk 3]
```diff
@@ -1,9 +1,92 @@
...
+| `output_dir` | The directory (folder) in which output should be saved. | `./output` | Path. |
```
Comment (MaxGamill-Sheffield) [inline docs/configuration.md:52]: "outputs" or "the output" (sorry to be pedantic)

File: docs/configuration.md [hunk 4]
```diff
@@ -1,9 +1,92 @@
...
+| `save_plots` | Should plots be saved? | `true` | `true` or `false` |
```
Comment (MaxGamill-Sheffield) [inline docs/configuration.md:53]: This and line 90 seem very similar. This is good for now but maybe we need to think about condensing the config file and avoid confusion in the future, and check what each of these do?

File: docs/configuration.md [hunk 5]
```diff
@@ -1,9 +1,92 @@
...
+
+### Filter
+
+The filter block contains one [threshold block](#threshold).
+
+### Grains
+
+| Parameter | Description | Example | Allowable |
+| --- | --- | --- | --- |
+| `gaussian_size` | The standard deviation of the Gaussian kernel used to filter grain sizes | 2.0 | Any positive real number. |
```
Comment (MaxGamill-Sheffield) [inline docs/configuration.md:69]: is the standard deviation value in pixels or nm?

File: docs/configuration.md [hunk 6]
```diff
@@ -1,9 +1,92 @@
...
+| `gaussian_mode` | *TBD* | nearest | *TBD* |
+| `absolute_smallest_grain_size` | The smallest grain size for analysis (pixels?). | 100 | Positive integer. |
+| `background` | *TBD* | `0.0` | *TBD* |
+| `zrange` | The height of the the highest and lowest objects under analysis. | [0, 3] | A list of two real numbers. |
```
Comment (MaxGamill-Sheffield) [inline docs/configuration.md:73]: I think this makes it sound like the user should be guessing the heights of things in their images. 
Maybe "Assigns values outside the low and high boundaries to the boundary values (purely aesthetic)".

File: docs/configuration.md [hunk 7]
```diff
@@ -1,9 +1,92 @@
...
+| `mask_direction` | *TBD* | `upper` | String `upper` or `lower`. |
```
Comment (MaxGamill-Sheffield) [inline docs/configuration.md:74]: "Direction to apply masking, above the surface or below the surface." maybe?

**Review by MaxGamill-Sheffield — CHANGES_REQUESTED** [review]
Some changes with wording but otherwise it looks good. 

Is there also a way to see this with the markdown applied?

**Comment by bobturneruk** [comment]
Thanks for the feedback @MaxGamill-Sheffield - the "rich diff" may be what you're looking for to see the document rendered as markdown https://github.com/AFM-SPM/TopoStats/pull/200/files?short_path=17ed184#diff-17ed18489a956f326ec0fe4040850c5bc9261d4631fb42da4c52891d74a59180

The docs can be built locally using:
```
pip install -e .[docs]
cd docs
make html
```

I'll have a look at the specific feedback soon.

**Comment by bobturneruk** [comment]
I suggest work here is paused until progress is made on https://github.com/AFM-SPM/TopoStats/issues/255

**Comment by ns-rse** [comment]
#255 has been resolved and #307 includes another attempt at documenting the configuration file options so I'm closing this PR.

It will be important as we progress that this documentation is kept up-to-date with changes in the code.

**Pull request closed without merging**